TypeCobol/Compiler/Parser/InspectedProgramClassDocument.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Collections.Generic;
+using TypeCobol.Compiler.Diagnostics;
+using TypeCobol.Compiler.Nodes;
+
+namespace TypeCobol.Compiler.Parser
+{
+    /// <summary>
+    /// Capture results of code analysis.
+    /// </summary>
+    public class InspectedProgramClassDocument
+    {
+        public InspectedProgramClassDocument(ProgramClassDocument programClassDocument, List<Diagnostic> violations)
+        {
+            PreviousStepSnapshot = programClassDocument;
+            Violations = violations;
+        }
+
+        /// <summary>
+        /// Snapshot of the fully parsed program that was quality checked
+        /// </summary>
+        public ProgramClassDocument PreviousStepSnapshot { get; }
+
+        /// <summary>
+        /// List of rules violations returned by code analysis
+        /// </summary>
+        public List<Diagnostic> Violations { get; }
```

Comment:
After discussion, can you rename Violations to Diagnostics and remove:
`public List<Diagnostic> Diagnostics => PreviousStepSnapshot.Diagnostics;`

Currently each Document return only its own Diagnostics.

We need to discuss what a Violation is. We can start by using Diagnostics to return violation of quality rules.

Reply:
Done with 0c68ca479d644a16680a4f146e5a39c2de6dd933.